### PR TITLE
Add cypress tests for Gateway-Policy feature UI

### PR DESCRIPTION
**Pull request by YasasRangika**
### Purpose
To enhance the quality and reliability of the Gateway Policy feature by introducing UI tests. These UI tests are designed to validate the functionality, user interactions, and edge cases within the Gateway Policy feature's user interface

### Goal
Test the UI implementations related to: https://github.com/wso2/api-manager/issues/1710

### Description
image

### CODE REVIEW

**Review by piyumaldk — CHANGES_REQUESTED** [review]

File: tests/cypress/integration/publisher/022-gateway-policies/00-gateway-policy.spec.js [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        const gateway = 'Default';
+        cy.get(`#gateway-chip-${gateway} svg[class*=MuiChip-deleteIcon]`)
+            .should('be.visible')
+            .click();
+        cy.get('[data-testid=undeploy-from-gateway-button]', { timeout: Cypress.config().largeTimeout }).click();
+
+        //Successful deletion of policy mapping
+        cy.visit(`publisher/global-policies`, { timeout: Cypress.config().largeTimeout });
+        cy.get('[data-testid=policy-mapping-delete-button]').click();
+        cy.get('[data-testid=policy-mapping-delete-confirmation-button]', { timeout: Cypress.config().largeTimeout }).click();
```
[the remark this answers is not in the document]
Comment (YasasRangika) [reply tests/cypress/integration/publisher/022-gateway-policies/00-gateway-policy.spec.js:99]: added with https://github.com/wso2/apim-apps/pull/526/commits/98518a7d447f70065b78a706a81e6d39ac4e1b0a.
Thank you!

File: tests/cypress/integration/publisher/022-gateway-policies/00-gateway-policy.spec.js [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+        cy.get('[data-testid=policy-attached-details-save]').should('be.disabled');
+        const newText = 'RequestHeaderValueNew';
+        cy.get('#headerValue').clear().type(newText).should('have.value', newText);
+        cy.get('[data-testid=policy-attached-details-save]').should('not.be.disabled');
+        cy.get('[data-testid=policy-attached-details-save]').click();
+        cy.get('[data-testid=policy-mapping-save-or-update-button]')
+            .scrollIntoView()
+            .should('be.visible')
+            .click();
+        cy.wait(2000);
```
[the remark this answers is not in the document]
Comment (YasasRangika) [reply tests/cypress/integration/publisher/022-gateway-policies/00-gateway-policy.spec.js:80]: added with https://github.com/wso2/apim-apps/pull/526/commits/98518a7d447f70065b78a706a81e6d39ac4e1b0a.
Thank you!

**Comment by YasasRangika** [comment]
After fixing latest review comments:
image

**Review by piyumaldk — DISMISSED** [review]

**Comment by YasasRangika** [comment]
For Reviewers, please note that the newly added test case is passing:
image

**Review by piyumaldk — APPROVED** [review]

**Review by dulithsenanayake — APPROVED** [review]
